libusb/core.c

```diff
@@ -2528,6 +2525,8 @@ int API_EXPORTED libusb_init_context(libusb_context **ctx, const struct libusb_i
 		*ctx = _ctx;
 
 		if (!usbi_fallback_context) {
+			if (usbi_atomic_load(&default_debug_level) == -1)
```

Comment:
As to validity of the sequence of the atomic operations, it works because this code is protected by the mutex.

Reply:
> When `defined(ENABLE_LOGGING) && !defined(ENABLE_DEBUG_LOGGING)` is `false`, neither `ctx_->debug` nor `default_debug_level` exist.

OK, I can see this is the case now with what is merged into master (there is a `#if defined(ENABLE_LOGGING) && !defined(ENABLE_DEBUG_LOGGING)` right below line 2527,  but as you can see, this line wasn't there when I did my comment, and, apart if for some reason I am not seeing your last version, I think it is not in your commit.

Anyway, I am happy with what is merged into master, thanks for taking the time to reply. 

I don't understand how this commit transformed into what is now merged into master.

> As to validity of the sequence of the atomic operations, it works because this code is protected by the mutex.

OK, thanks I missed that.